projects/plugins/jetpack/_inc/client/state/site/reducer.js

```diff
@@ -436,13 +437,17 @@ export function getActiveVideoPressPurchase( state ) {
 }
 
 /**
- * Determines if the site has an active VideoPress product purchase
+ * Determines if the site has an active VideoPress feature and excludes the VideoPress feature from Security plans
+ * purchased after a certain date.
  *
  * @param {*} state - Global state tree
  * @returns {boolean} True if the site has an active VideoPress product purchase, false otherwise.
  */
-export function hasActiveVideoPressPurchase( state ) {
-	return !! getActiveVideoPressPurchase( state );
+export function hasActiveVideoPressFeature( state ) {
+	return (
+		hasActiveSiteFeature( state, 'videopress' ) &&
+		! getSitePurchases( state ).find( isSecurityPlanExcludingVideoPress )
```

Comment:
This needs to also support if the customer has a "Security plan excluding VideoPress" plan but also separately purchased the VideoPress product. It looks like this scenario is not providing the progress bar (currently a bug in prod as well it seems) and is also triggering the "Free trial" ui on this PR.

<img width="1017" alt="Screen Shot 2022-05-02 at 11 51 28 AM" src="https://user-images.githubusercontent.com/4081020/166265148-996e26d7-cd03-440c-b362-5be93f632cba.png">
<img width="814" alt="Screen Shot 2022-05-02 at 11 51 38 AM" src="https://user-images.githubusercontent.com/4081020/166265151-81cdf910-581d-4a53-86c5-921ed1c06fcb.png">

Reply:
Good catch! -- I don't think the || will solve this unfortunately. We talked a bit in #limited-atomic p1651506941639029-slack-C02V49GM0PN about solving this issue by having `hasActiveSiteFeature` support legacy purchases. I think once that's done, that should clear up this particular problem, and I'll update this PR. It will also centralize the legacy purchases definitions, which will be nice. 😅

Reply:
I'll keep an eye on this then, because any changes your team is making to the endpoint might help us clean up some of the special case code we have on the video upload side for storage checks as well.

Reply:
Hi, @jgcaruso! 👋 We finally implemented legacy features in D80783-code and I've updated this PR to use them. The features are defined in D80962-code. I believe that's fixed this particular issue. Might you have any feedback on this implementation?